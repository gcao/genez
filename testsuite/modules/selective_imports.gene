#!/usr/bin/env gene run

# Test selective imports - create a module in memory first
(ns utils
  (fn helper [x] (* x 2))
  (fn process [x] (+ x 10))
  ^VERSION 1.0)

# Test selective import of specific items
(import "utils" [helper VERSION])

# These should work
(print "Helper result: " (helper 5))
(print "Version: " VERSION)

# This should fail if selective imports work correctly
# (print "Process result: " (process 5))  # Should error - not imported