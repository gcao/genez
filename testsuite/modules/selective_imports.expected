Helper result:  10
Version:  1.0